lib/viewRenderer/nodeRendering/NodeBuilder.ts

```diff
@@ -0,0 +1,226 @@
+  modelElementId: string;
+  viewNodeId: string;
+  posX?: number;
+  posY?: number;
+  parentElement?: dia.Cell | null;
+}
+
+/**
+ * Class with functionality to build a node
+ *
+ * @example
+ * import { dia } from 'jointjs';
+ * import { NodeBuilder } from '@lib/viewRenderer/nodeRendering/NodeBuilder';
+ * import { ViewSettings } from '@lib/viewRenderer/ViewSettings';
+ *
+ * const graph = new dia.Graph({ cellNamespace: shapes });
+ * const nodeBuilder = new NodeBuilder(graph, new ViewSettings({}));
+ */
+export class NodeBuilder {
+  private readonly style: Style;
+  private builder: ShapeBuilder;
+
+  /**
+   *
+   * @param graph Jointjs Graph model
+   * @param settings Node view settings
+   */
+  constructor(private graph: dia.Graph, settings: ViewSetting) {
+    const { style } = settings;
+
+    this.style = style;
+    this.builder = new ShapeBuilder(settings);
+  }
+
+  /**
+   * Builds a node shape based on the node classification type and other given attributes
+   * @param attributes
+   * @param attributes.name Node name
+   * @param attributes.type Node type
+   * @param attributes.width Node width
+   * @param attributes.height Node height
+   * @return Node shape
+   * @example
+   * import { dia } from 'jointjs';
+   * import { NodeBuilder } from '@lib/viewRenderer/nodeRendering/NodeBuilder';
+   * import { ViewSettings } from '@lib/viewRenderer/ViewSettings';
+   *
+   * const graph = new dia.Graph({ cellNamespace: shapes });
+   * const nodeBuilder = new NodeBuilder(graph, new ViewSettings({}));
+   * const rectangularShape = nodeBuilder.buildShape({
+   *         type: 'applicationcomponent',
+   *         name: 'Model Service',
+   *         width: 100,
+   *         height: 80,
+   * })
+   */
+  buildShape({ name, type, width, height }: BasicNodeAttributes) {
+    const buildBasicRectangular = (nodeType, attributes) => {
+      return this.builder.buildBasicRectangular(name, {
+        width,
+        height,
+        fillColor: typeToHexColor(nodeType, this.style),
+        ...attributes,
+      });
+    };
+
+    const shapeClassification = {
+      [NodeShapeClassification.Structure]: () => buildBasicRectangular(type, {}),
+      [NodeShapeClassification.Behaviour]: () =>
+        this.builder.buildBasicRounded(name, {
+          width,
+          height,
+          fillColor: typeToHexColor(type, this.style),
+        }),
+      [NodeShapeClassification.ImplementationAndMigration]: () =>
+        this.builder.buildBasicRounded(name, {
+          width,
+          height,
+          fillColor: typeToHexColor(type, this.style),
+        }),
+      [NodeShapeClassification.Motivational]: () =>
+        this.builder.buildBasicOctagonal(name, {
+          width,
+          height,
+          fillColor: typeToHexColor(type, this.style),
+        }),
+      [NodeType.Grouping]: () =>
+        buildBasicRectangular(type, {
+          withDashedStroke: true,
+          textAnchor: 'left',
```

Comment:
Hi, @rodrigolimoes . Do you think it would be a good idea to create enums or consts for these string literals (e.g. 'left', 'right', 'text'?

Reply:
Yes, I think it's a good idea to create an enumeration for position strings. Makes code easier to understand and easier to maintain.

`
export enum Position {
  Left = 'left',
  Right = 'right',
  Bottom = 'bottom',
  Top = 'top',
  Middle = 'middle'
}`

What do you think this enum?

Reply:
Hi, @rodrigolimoes . Great suggestion.

Reply:
I fixed it